model/cloud.go

```diff
@@ -46,6 +46,15 @@ type CloudCustomer struct {
 	PaymentMethod    *PaymentMethod `json:"payment_method"`
 }
 
+// CloudCustomerPatch represents a patch of a customer.
+type CloudCustomerPatch struct {
+	Name             string `json:"name"`
+	Email            string `json:"email"`
+	ContactFirstName string `json:"contact_first_name"`
+	ContactLastName  string `json:"contact_last_name"`
+	NumEmployees     int    `json:"num_employees"`
+}
+
```

Comment:
I think there's a good opporunity for a slight refactor here. Since this is basically entirely overlapping with the CloudCustomer type, could we do something like:
```
type CloudCustomerInfo struct {
	Name             string `json:"name"`
	Email            string `json:"email"`
	ContactFirstName string `json:"contact_first_name"`
	ContactLastName  string `json:"contact_last_name"`
	NumEmployees     int    `json:"num_employees"`
}
```
Then change the definition of CloudCustomer to include these fields by [embedding the CloudCustomerInfo struct](https://golang.org/doc/effective_go.html#embedding)
```
type CloudCustomer struct {
        CloudCustomerInfo // Embed the CloudCustomerInfo 
	ID               string         `json:"id"`
	CreatorID        string         `json:"creator_id"`
	CreateAt         int64          `json:"create_at"`
	BillingAddress   *Address       `json:"billing_address"`
	CompanyAddress   *Address       `json:"company_address"`
	PaymentMethod    *PaymentMethod `json:"payment_method"`
}
```

I believe this would maintain backwards compatibility and help with duplication of entities in the backend - but @gabrieljackson might be better to confirm this

Reply:
I'm on board with this as long as it doesn't complicate things any further.

Reply:
Yep! This should work without much issue at all.